vita3k/http/include/http/state.h

```diff
@@ -0,0 +1,266 @@
+// Vita3K emulator project
+// Copyright (C) 2022 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#pragma once
+
+#include <mem/ptr.h>
+
+#include <map>
+#include <string>
+#include <util/types.h>
+#include <vector>
+
+#define SCE_HTTP_DEFAULT_RESPONSE_HEADER_MAX 5000U
+#define SCE_HTTP_USERNAME_MAX_SIZE 256
+#define SCE_HTTP_PASSWORD_MAX_SIZE 256
+
+typedef int (*SceHttpsCallback)(unsigned int verifyEsrr, void *const sslCert[], int certNum, void *userArg);
+
+enum SceHttpsErrorCode {
+    SCE_HTTPS_ERROR_CERT = -2143072160,
+    SCE_HTTPS_ERROR_HANDSHAKE = -2143072159,
+    SCE_HTTPS_ERROR_IO = -2143072158,
+    SCE_HTTPS_ERROR_INTERNAL = -2143072157,
+    SCE_HTTPS_ERROR_PROXY = -2143072156
```

Comment:
Could you put the errors in hexadecimal form (like 0x80435060 for the first one)? it would be cleaner and like the other errors